extensions/warp-ipfs/src/store/message/community_task.rs

```diff
@@ -3304,29 +3306,125 @@ impl CommunityTask {
     }
     pub async fn attach_to_community_channel_message(
         &mut self,
-        _channel_id: Uuid,
-        _message_id: Option<Uuid>,
-        _locations: Vec<Location>,
-        _message: Vec<String>,
+        channel_id: Uuid,
+        message_id: Option<Uuid>,
+        locations: Vec<Location>,
+        messages: Vec<String>,
     ) -> Result<(Uuid, AttachmentEventStream), Error> {
-        Err(Error::Unimplemented)
+        let own_did = &self.identity.did_key();
+        if !self.document.has_channel_permission(
+            own_did,
+            &CommunityChannelPermission::SendMessages,
```

Comment:
I wonder if we should split this so it has two permission checks, one for sending messages and another for attaching a file. Thoughts?

Reply:
Sounds good. Done 👍